kyo-core/shared/src/main/scala/kyo/Fiber.scala

```diff
@@ -278,21 +278,17 @@ object Fiber extends FiberPlatformSpecific:
             end State
             val state = new State
             import state.*
-            foreach(seq)((idx, io) => io.evalNow.foreach(v => state(Result.success(v))))
-            if state.done() then
-                state
-            else
-                boundary { (trace, context) =>
-                    IO {
-                        foreach(seq) { (_, v) =>
-                            val fiber = IOTask(v, safepoint.copyTrace(trace), context)
-                            state.interrupts(fiber)
-                            fiber.onComplete(state)
-                        }
-                        state
+            boundary { (trace, context) =>
+                IO {
+                    val interruptPanic = Result.Panic(Fiber.Interrupted(frame))
+                    foreach(seq) { (_, v) =>
+                        val fiber = IOTask(v, safepoint.copyTrace(trace), context)
+                        state.onComplete(_ => discard(fiber.interrupt(interruptPanic)))
```

Comment:
This is the main fix. Instead of using `state.interrupts`, which would trigger only if the state promise itself gets interrupted, the fix uses `onComplete` to interrupt loser fibers as soon as the `state` is complete. If the fiber is already completed, `interrupt` will return `false`, which is ok to ignore.